src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -1625,6 +1627,78 @@ public void shouldNotChangeStatusUpdatedDateWhenStatusHasNotChanged() throws Exc
       .getString(STATUS_UPDATED_DATE_PROPERTY), withinSecondsBeforeNow(seconds(2)));
   }
 
+  private void assertExists(JsonObject expectedInstance) throws Exception {
+    Response response = getById(expectedInstance.getString("id"));
+    assertThat(response, statusCodeIs(HttpStatus.HTTP_OK));
+    assertThat(response.getBody(), containsString(expectedInstance.getString("title")));
+  }
+
+  @Test
+  public void canSyncPost() throws Exception {
+    JsonArray instancesArray = new JsonArray();
+    int numberOfInstances = 1000;
+
+    instancesArray.add(uprooted(UUID.randomUUID()));
+    for(int i = 2; i < numberOfInstances; i++) {
+      instancesArray.add(smallAngryPlanet(UUID.randomUUID()));
+    }
+    instancesArray.add(temeraire(UUID.randomUUID()));
+
+    JsonObject instanceCollection = new JsonObject().put(INSTANCES_KEY, instancesArray);
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+    client.post(instancesStorageSyncUrl(""), instanceCollection, TENANT_ID, ResponseHandler.empty(createCompleted));
+    Response response = createCompleted.get(5, SECONDS);
+    assertThat(response, statusCodeIs(HttpStatus.HTTP_CREATED));
+
+    JsonObject uprooted         = instancesArray.getJsonObject(0);
+    JsonObject smallAngryPlanet = instancesArray.getJsonObject(numberOfInstances / 2);
+    JsonObject temeraire        = instancesArray.getJsonObject(numberOfInstances - 1);
+    assertExists(uprooted);
+    assertExists(smallAngryPlanet);
+    assertExists(temeraire);
+  }
+
+  @Test
+  public void cannotSyncPostWithInvalidInstance() throws Exception {
+    JsonArray instancesArray = new JsonArray();
+    instancesArray.add(uprooted(UUID.randomUUID()));
+    instancesArray.add(smallAngryPlanet(UUID.randomUUID()).put("foo", "bar"));  // "foo" is invalid
```

Comment:
I'd extract the `"foo"` to a variable (e.g. `invalidPropertyName`), so there will be no need for the comment.

Reply:
Good point, I've replaced "foo" by "invalidPropertyName" and removed the comment.
I don't think that extracting the String into a variable improves readability.